component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/dao/JWTStorageManager.java

```diff
@@ -28,154 +28,194 @@
  */
 public class JWTStorageManager {
 
     private static final Log log = LogFactory.getLog(JWTStorageManager.class);
 
     /**
-     * Check whether a JWT Entry with given jti exists in the DB.
+     * To get a list of persisted JWTs for a given JTI.
      *
-     * @param jti JWT token id
-     * @return true if an entry is found
-     * @throws IdentityOAuth2Exception when exception occurs
-     */
-    public boolean isJTIExistsInDB(String jti) throws OAuthClientAuthnException {
-
-        Connection dbConnection = IdentityDatabaseUtil.getDBConnection();
-        PreparedStatement prepStmt = null;
-        boolean isExists = false;
-        ResultSet rs = null;
-        try {
-            prepStmt = dbConnection.prepareStatement(SQLQueries.GET_JWT_ID);
-            prepStmt.setString(1, jti);
-            rs = prepStmt.executeQuery();
-            int count = 0;
-            if (rs.next()) {
-                count = rs.getInt(1);
-            }
-            if (count > 0) {
-                isExists = true;
-            }
-        } catch (SQLException e) {
-            if (log.isDebugEnabled()) {
-                log.debug("Error when retrieving the JWT ID: " + jti, e);
-            }
-            throw new OAuthClientAuthnException("Error occurred while validating the JTI: " + jti + " of the " +
-                                                "assertion.", OAuth2ErrorCodes.INVALID_REQUEST);
-        } finally {
-            IdentityDatabaseUtil.closeAllConnections(dbConnection, rs, prepStmt);
-        }
-        return isExists;
-    }
-
-    /**
-     * To get persisted JWT for a given JTI.
-     *
-     * @param jti jti
-     * @return JWTEntry
+     * @param jti JTI.
+     * @return List of JWTEntries.
      * @throws OAuthClientAuthnException OAuthClientAuthnException thrown with Invalid Request error code.
      */
-    public JWTEntry getJwtFromDB(String jti) throws OAuthClientAuthnException {
+    public List<JWTEntry> getJwtsFromDB(String jti, int tenantId) throws OAuthClientAuthnException {
+
+        List<JWTEntry> JWTEntries = new ArrayList<>();
 
         Connection dbConnection = IdentityDatabaseUtil.getDBConnection();
         PreparedStatement prepStmt = null;
         ResultSet rs = null;
-        JWTEntry jwtEntry = null;
         try {
-            prepStmt = dbConnection.prepareStatement(SQLQueries.GET_JWT);
-            prepStmt.setString(1, jti);
-            rs = prepStmt.executeQuery();
-            if (rs.next()) {
-                long exp = rs.getTime(1, Calendar.getInstance(TimeZone.getTimeZone(Constants.UTC))).getTime();
-                long created = rs.getTime(2, Calendar.getInstance(TimeZone.getTimeZone(Constants.UTC))).getTime();
-                jwtEntry = new JWTEntry(exp, created);
+            if (Util.isIsTenantIdColumnIsAvailableInIdnOidcAuthTable()){
+                prepStmt = dbConnection.prepareStatement(Util.getDBQuery(GET_JWT_WITH_DEFAULT_TENANT));
```

Comment:
From the query 'GET_JWT_WITH_DEFAULT_TENANT' it will return the jwts with both default tenant id and any other tenant ids right? So we better rename the constant.